src/core/libmaven/mzAligner.cpp

```diff
@@ -452,26 +452,31 @@ bool Aligner::alignSampleRts(mzSample* sample,
                              bool setAsReference,
                              const MavenParameters* mp)
 {
-    vector<float> rtPoints(sample->scans.size());
-    vector<vector<float> > mxn(sample->scans.size());
-    for (int j = 0; j < sample->scans.size(); ++j) {
-        if (mp->stop) return (true);
-        rtPoints[j] = sample->scans[j]->originalRt;
-        mxn[j] = vector<float> (mzPoints.size());
+    vector<float> rtPoints;
+    vector<vector<float> > mxn;
+
+    int mxnCount = 0;
+    for(auto scan: sample->scans) {
+        if(mp->stop) return (true);
+        if(scan->mslevel == 1 ) {
+            rtPoints.push_back(scan->originalRt);
+            mxn.push_back(vector<float> (mzPoints.size()));
+        }
     }
 
-    for (int j = 0 ; j < sample->scans.size(); ++j) {
-        for (int k = 0; k < sample->scans[j]->mz.size(); ++k) {
-            if (mp->stop) return (true);
-            if (sample->scans[j]->mz[k] < mzPoints.front()
-                || sample->scans[j]->mz[k] > mzPoints.back())
-                continue;
-            int index = upper_bound(mzPoints.begin(), mzPoints.end(), sample->scans[j]->mz[k])
-                            - mzPoints.begin() -1;
-
-            mxn[j][index] = max(mxn[j][index], sample->scans[j]->intensity[k]);
+    for(auto scan: sample->scans) {
+        if(scan->mslevel == 1) {
+            mxnCount++;
+            for(int indice = 0; indice <  scan->mz.size(); indice++) {
```

Comment:
Did you mean `index` instead of `indice`? Also, we've agreed to single letter counter variables in our style guide, so we can just have this as `i`, considering `index` name is being used by a variable declared within this for-loop block.